weave/trace/weave_client.py

```diff
@@ -1042,6 +1059,95 @@ def query_costs(
+            scorer_call_ref_uri = scorer_call_ref.uri()
+            scorer_op_ref_uri = score_call.op_name
+            scorer_op_ref = parse_uri(scorer_op_ref_uri)
+            if not isinstance(scorer_op_ref, OpRef):
+                raise ValueError(f"Invalid scorer op ref: {scorer_op_ref_uri}")
+            score_name = scorer_op_ref.name
+            score_results = score_call.output
+
+            return self._add_score(
+                call_ref_uri=call_ref_uri,
+                score_name=score_name,
+                score_results=score_results,
+                scorer_call_ref_uri=scorer_call_ref_uri,
+                scorer_op_ref_uri=scorer_op_ref_uri,
+            )
+
+        return self.future_executor.defer(send_score_call)
+
+    @trace_sentry.global_trace_sentry.watch()
+    def _add_score(
+        self,
+        *,
+        call_ref_uri: str,
+        score_name: str,
+        score_results: Any,
+        scorer_call_ref_uri: str,
+        scorer_op_ref_uri: str,
+    ) -> str:  # , supervision: dict):
+        """(Private) Low-level, non object-oriented method for adding a score to a call.
+
+        Outstanding questions:
+        - Should we somehow include supervision (ie. the ground truth) in the payload?
+        - What should the shape of `ScoreTypePayload` be? Maybe we want the results to be top-level?
+        - What should we use for name? A standard "score" or the score name?
+        """
+        # Parse the refs (acts as validation)
+        call_ref = parse_uri(call_ref_uri)
+        if not isinstance(call_ref, CallRef):
+            raise ValueError(f"Invalid call ref: {call_ref_uri}")
+        scorer_call_ref = parse_uri(scorer_call_ref_uri)
+        if not isinstance(scorer_call_ref, CallRef):
+            raise ValueError(f"Invalid scorer call ref: {scorer_call_ref_uri}")
+        scorer_op_ref = parse_uri(scorer_op_ref_uri)
+        if not isinstance(scorer_op_ref, OpRef):
+            raise ValueError(f"Invalid scorer op ref: {scorer_op_ref_uri}")
+
+        # Validate score_name (we might want to relax this in the future)
+        if score_name != scorer_op_ref.name:
+            raise ValueError(
+                f"Score name {score_name} does not match scorer op name {scorer_op_ref.name}"
+            )
+
+        # Prepare the result payload - we purposely do not map to refs here
+        # because we prefer to have the raw data.
+        results_json = to_json(score_results, self._project_id(), self)
+
+        # # Prepare the supervision payload
+
+        payload: ScoreTypePayload = {
+            "name": score_name,
+            "op_ref": scorer_op_ref_uri,
+            "call_ref": scorer_call_ref_uri,
+            "results": results_json,
+        }
+
+        freq = FeedbackCreateReq(
+            project_id=self._project_id(),
+            weave_ref=call_ref_uri,
+            feedback_type=SCORE_TYPE_NAME,  # should this be score_name?
+            payload=payload,
```

Comment:
is this payload arbitrary and you are just defining a specific type of payload of shape `ScoreTypePayload`?

Reply:
yup!